app/src/androidTest/java/org/simple/clinic/encounter/EncounterRepositoryAndroidTest.kt

```diff
@@ -99,4 +102,106 @@ class EncounterRepositoryAndroidTest {
       assertThat(updatedAt).isEqualTo(Instant.parse("2018-02-13T00:00:00Z"))
     }
   }
+
+  @Test
+  fun when_a_bp_is_saved_encounter_should_get_created() {
+    //given
+    val patientUuid = UUID.fromString("6f725ffd-7008-4018-9e7c-33346964a0c1")
+    val facilityUuid = UUID.fromString("22bdeedb-061e-4a17-8739-e946a4206593")
+    val bpUuid = UUID.fromString("2edd4c06-e2de-4a18-a8e7-43e2e30c9aba")
+    val encounteredDate = LocalDate.parse("2018-01-01")
+    val bloodPressureMeasurement = testData.bloodPressureMeasurement(
+        uuid = bpUuid,
+        patientUuid = patientUuid,
+        facilityUuid = facilityUuid,
+        recordedAt = Instant.parse("2018-01-01T00:00:00Z"),
```

Comment:
Curious, will the `encounteredDate` be the same as `recordedAt`?

Reply:
Yes. `encounteredDate` is derived from `recordedAt`.
There is no way to set the `encounterDate` by the user. Also the `encounterUuid` itself is generated using the `encounterDate` itself. This is not enforced through the APIs itself but it is handled in the `EncounterRepository` itself. Refer `saveBloodPressureMeasurement` method in `EncounterRepository`.